wrappers/desktop.go

```diff
@@ -104,13 +104,34 @@ var isValidDesktopFileLine = regexp.MustCompile(strings.Join([]string{
 	"^TargetEnvironment=",
 }, "|")).Match
 
-// rewriteExecLine rewrites a "Exec=" line to use the wrapper path for snap application.
-func rewriteExecLine(s *snap.Info, desktopFile, line string) (string, error) {
-	env := fmt.Sprintf("env BAMF_DESKTOP_FILE_HINT=%s ", desktopFile)
+// fileOrUriMacro returns the Exec line macro used to expand files or URIs
+func fileOrUriMacro(cmd string) rune {
+	isMacro := false
+	for _, r := range cmd {
+		if isMacro {
+			switch r {
+			case 'f', 'F', 'u', 'U':
+				return r
+			}
+			isMacro = false
+		} else if r == '%' {
+			isMacro = true
+		}
+	}
+	// If no macros are found, default to accepting a single regular file
+	return 'f'
+}
 
+// rewriteExecLine rewrites a "Exec=" line to use the wrapper path for snap application.
+func rewriteExecLine(s *snap.Info, desktopFile, action, line string) (string, error) {
+	if action != "" {
+		action = " --action " + action
+	}
 	cmd := strings.SplitN(line, "=", 2)[1]
+	exec := fmt.Sprintf("Exec=/usr/bin/snap routine desktop-launch --desktop %s%s -- %%%c\nX-Snap-Exec=", desktopFile, action, fileOrUriMacro(cmd))
```

Comment:
I wonder if instead of this, having something like:

return fmt.Sprinf("Exec=%s\nX-Snap-Exec=%s..."

below when we construct the results would be slightly more readable or make no difference?